### PR TITLE
ref: project generation

### LINKED ISSUE #37: Use GeneratorNewProjectWizard for project creation
Current approach using `ModuleBuilder` does not create nice, single step project wizard view:

image

Plugin might use `GeneratorNewProjectWizard` to create with single wizard step like:

image

To achieve above project wizard must contain:
- project name input
- project location input
- jdk selection

All of above is already implemented in [CommonStarterInitialStep.java](https://github.com/JetBrains/intellij-community/blob/867f23c3b356d6983bf14dc9cb4a4432988ca1d7/java/idea-ui/src/com/intellij/ide/starters/shared/CommonStarterInitialStep.kt#L4) but it's not available in IntelliJ API.

### PR DESCRIPTION
**Pull request by MarcinVaadin**
## Description

New Project generation instead of Module generation

Fixes #37

image

### CODE REVIEW

File: src/main/kotlin/com/vaadin/plugin/module/VaadinPanel.kt
Comment (MarcinVaadin) [inline src/main/kotlin/com/vaadin/plugin/module/VaadinPanel.kt:1]: Most changes in this file copied from https://github.com/JetBrains/intellij-community/blob/2a49997971c83e6f195c139f15d770015a55d290/java/idea-ui/src/com/intellij/ide/starters/shared/CommonStarterInitialStep.kt#L4

File: src/main/kotlin/com/vaadin/plugin/module/VaadinProjectBuilderAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+            projectDownloadedProperty.afterChange { afterProjectCreated(project) }
+            VaadinProjectUtil.downloadAndExtract(project, vaadinWizard.projectModel!!.getDownloadLink(project))
+            project
+        }
+    }
+
+    override fun isAvailable(): Boolean {
+        val lastPerformedActionId = (ActionManagerImpl.getInstance() as ActionManagerImpl).lastPreformedActionId
+        lastPerformedActionId ?: return false
+        return lastPerformedActionId.contains("NewProject", true)
```
Comment (MarcinVaadin) [inline src/main/kotlin/com/vaadin/plugin/module/VaadinProjectBuilderAdapter.kt:39]: Show Vaadin wizard only on New Project creation, not New Module.

**Review by abdullahtellioglu — CHANGES_REQUESTED** [review]
Hello world projects tab does not work. Plugin creates the project only by first section's configurations. 

https://github.com/vaadin/intellij-plugin/assets/100126447/5fc814a9-8050-43cb-a88f-fd2ad5887558

File: src/main/kotlin/com/vaadin/plugin/module/VaadinProjectBuilderAdapter.kt (hunk 1 above)
Comment (abdullahtellioglu) [reply src/main/kotlin/com/vaadin/plugin/module/VaadinProjectBuilderAdapter.kt:39]: This is OK, the new module window does not have Vaadin section

**Comment by MarcinVaadin** [comment]
> Hello world projects tab does not work. Plugin creates the project only by first section's configurations.

Fixed, good catch

**Review by abdullahtellioglu — APPROVED** [review]
All works!

**Pull request merged**